src/server/cea_608_encoder/cea_608_utility.py

```diff
@@ -56,19 +56,69 @@ def bytes_to_byte_pairs(byte_list: list) -> list:
     return byte_pairs
 
 
-def create_byte_pairs(caption_string: str) -> list:
+def which_char_set(caption_char: str) -> str:
+    """Finds which character set the letter is in
+
+    :param caption_char:
+    :return: the name of the char set the caption_char is in
+    """
+    for set_name,set in char_sets.items():
+        for letter,decimal in set.items():
+            if caption_char == letter:
+                return set_name
+    raise ValueError(f'The character: {caption_char} is not in any of the supported character sets')
```

Comment:
We have this handled in your changes in `which_channel` we can probably remove this one

Reply:
What do you suggest for if the character is not in any of the sets then?

Reply:
if it's not in any, then when we can `which_channel` there will be an error raised